src/test_l1_dsAudio.c

```diff
@@ -99,6 +100,106 @@ intptr_t null_handle = 0;
 
 /* Global flags to support features */
 static bool extendedEnumsSupported=false; //Default to not supported
+static char* device_type = NULL;
+static char* device_name = NULL;
+static int numports = 0;
+
+typedef struct  port_config {
+	uint16_t typeid;
+	uint16_t index;
+	char* port_name;
+	int no_of_supported_compression;
+	int supported_compressions[ARRAY_SIZE];
+	int no_of_supported_encodings;
+	int supported_encodings[ARRAY_SIZE];
+	int no_of_supported_stereo_mode;
+	int supported_stereo_mode[ARRAY_SIZE];
+	int ms12_capabilites;
+	char* ms12_audio_profiles[ARRAY_SIZE];
```

Comment:
this array of pointers and you need to allocate memory for the pointers, otherwise code will crash. Better to use 2D array here